### PR TITLE
[JENKINS-68890] fix dual package compiler error detected by eclipse compiler

### PR DESCRIPTION
**Pull request by jtnord**
See [JENKINS-68890](https://issues.jenkins.io/browse/JENKINS-68890).

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since="TODO")` or `@Deprecated(since="TODO", forRemoval=true)` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6722"></a>

### CODE REVIEW

File: test/pom.xml [hunk 1]
```diff
@@ -97,6 +97,10 @@ THE SOFTWARE.
           <groupId>${project.groupId}</groupId>
           <artifactId>jenkins-war</artifactId>
         </exclusion>
+        <exclusion>
```
Comment (timja) [inline test/pom.xml:100]: better to fix upstream? easy to quickly integrate fix

Comment (Vlatombe) [reply test/pom.xml:100]: was about to say the same but jth should still be java 8 compatible for now right?

Comment (timja) [reply test/pom.xml:100]: currently till LTS I believe, if there's no fix we can do to it test harness I think we should file an issue against it and a TODO comment here?

Comment (jtnord) [reply test/pom.xml:100]: This is at least my qucik fix whilst I focus on getting a build of Jenkins so I can actually debug https://github.com/jenkinsci/jenkins/pull/6715#pullrequestreview-1023847744

jaxb was bundled with java8 so we may well be able to exclude it across the board.  (no idea how / if this impacts java17 as of yet though).  Will leave this here until I finish with the other issue.

**Comment by basil** [comment]
Closing in favor of #6775.

**Pull request closed without merging**